### ISSUE TITLE
[a11y]: SidePanel should close on `escape` key

### ISSUE THREAD

**Issue by matthewgallo** [carbon-design-system/ibm-products#5765]
### Package

@carbon/ibm-products

### Browser

Firefox

### Operating System

Windows

### Package version

`2.46.0-rc.0`

### Automated testing tool and ruleset

Manual

### Assistive technology

JAWS

### Description

Since in most cases the `SidePanel` is a dialog, the <kdb>escape</kbd> should close the panel when pressed. The only exception should be for `slideIn` panels, as they are not dialogs and do not trap focus.

### WCAG 2.1 Violation

4.1.2 Name Role Value

### Reproduction/example

https://ibm-products.carbondesignsystem.com/?path=/story/ibm-products-components-side-panel-sidepanel--slide-over&globals=viewport:basic

### Steps to reproduce

Visit [side panel story](https://ibm-products.carbondesignsystem.com/?path=/story/ibm-products-components-side-panel-sidepanel--slide-over&globals=viewport:basic), attempt close panel with <kdb>escape</kdb>.

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/ibm-products/blob/main/.github/CODE_OF_CONDUCT.md)
- [X] I checked the [current issues](https://github.com/carbon-design-system/ibm-products/issues) for duplicate problems
```[tasklist]
### Tasks
- [ ] Side panel can close with `escape` key, with the exception of `slideIn` side panels
```
